src/com/palmergames/bukkit/towny/db/TownyDatabaseHandler.java

```diff
@@ -626,32 +626,25 @@ public void removeResident(Resident resident) {
 
 	@Override
 	public void removeTownBlock(TownBlock townBlock) {
+		Town town = null;
+		try {
+			town = townBlock.getTown();
+		} catch (NotRegisteredException e) {
+			// TODO: Log why a TownBlock would not have a town.
+			// If there is not an associated town, it's probably not a valid TownBlock. Should we return?
```

Comment:
if it has no valid town it should be nuked still (I don't think townless townblocks are possible anyways nowadays.)

Reply:
Should now log as an error, with an early return.

Reply:
By nuking, I meant it should be removed. Any townblock missing a town has to get removed otherwise we end up with ghost-townblocks. The return can be removed from here, I've made a separate comment about the TownPreUnclaimEvent having the town annotated as well.